### DISCUSSION TITLE
Behind the corporate firewall

### DISCUSSION THREAD

**Discussion by oravecz** [leoafarias/fvm#571] [Q&A]
We cannot install Flutter from a Git cache, nor can we download Flutter from the `https://storage.googleapis.com/flutter_infra_release/releases` location. Instead, we maintain our own versions of Flutter SDKs on an internal location which we set our `FLUTTER_STORAGE_BASE_URL` to. We also maintain our own version of the Flutter metadata files with our own checksum values in the same manner as Google.

Looking at the source code, I thought that FVM would respect the `FLUTTER_STORAGE_BASE_URL` environment variable, but it does not.

Is it feasible to make a modification to the source code to ensure that `FLUTTER_STORAGE_BASE_URL` is used as the source for Flutter packages to be put in the local cache?

**Comment by oravecz** [comment]
I think my discussion might be a bit off-topic, because i see you are not downloading a pre-packaged version of Flutter, but rather accessing the Flutter Git repo directly. Unfortunately, this won't pass our firm's control procedures, and we will have to take a different approach.

**Reply by leoafarias** [reply]
@oravecz Sorry for the delay in responding, but it appears I initially missed the notification. The FLUTTER_STORAGE_BASE_URL is primarily for versions, and it does not facilitate the download of the Flutter version zip with dependencies. This is a feature I would like to see implemented in the future. Could you please create an issue for this enhancement?

**Comment by oravecz** [comment]
I created #688, and you may have a specific intent for it.

For us, it proved to be a bit of a red herring. Google does not let anyone outside of the China CDN efficiently mirror their repositories. By this I mean, they do not allow browsing of folder structures, nor do they publish a repository manifest to allow someone to know which files are available for mirroring. Because each instance of Dart and Flutter use different, dynamic paths, and these paths also differ by chipset/architecture, a manifest cannot be reliably created without Google's assistance.

Because of this, we stopped trying to do so, and moved to an approach where we maintain an internal fork of the Flutter repository with only those versions our internal teams have approved. To continue using FVM to manage our automated build nodes and local developer Flutter versions per project, we are leveraging the support you recently committed in PR #593 to allow the FLUTTER_GIT_URL to be supported within FVM. Without this, we were nearly dead in the water.
